atc/api/cc_test.go

```diff
@@ -355,10 +355,48 @@ var _ = Describe("cc.xml", func() {
 		Context("when not authenticated", func() {
 			BeforeEach(func() {
 				fakeAccess.IsAuthenticatedReturns(false)
+				var fakeTeam *dbfakes.FakeTeam
+				fakeTeam = new(dbfakes.FakeTeam)
+				fakeTeam.NameReturns("a-public-team")
+				dbTeamFactory.FindTeamReturns(fakeTeam, true, nil)
+				fakeNonPublicPipeline := new(dbfakes.FakePipeline)
+				fakeTeam.PipelinesReturns([]db.Pipeline{
+					fakeNonPublicPipeline,
+				}, nil)
+
+				fakePublicPipeline := new(dbfakes.FakePipeline)
+				fakeTeam.PublicPipelinesReturns([]db.Pipeline{
+					fakePublicPipeline,
+				}, nil)
+				endTime, _ := time.Parse(time.RFC3339, "2018-11-04T21:26:38Z")
+				fakePublicPipeline.DashboardReturns([]atc.JobSummary{
+					{
+						Name:         "a-public-job",
+						PipelineName: "exposed-pipeline",
+						TeamName:     "a-public-team",
+						FinishedBuild: &atc.BuildSummary{
+							Name:    "43",
+							Status:  "succeeded",
+							EndTime: endTime.Unix(),
+						},
+					},
+				}, nil)
+
+			})
+
+			It("returns 200", func() {
+				Expect(response.StatusCode).To(Equal(http.StatusOK))
 			})
 
-			It("returns 401", func() {
-				Expect(response.StatusCode).To(Equal(http.StatusUnauthorized))
+			It("lists public pipelines", func() {
```

Comment:
This looks good! Adding another test for when there's no public pipelines will be a good idea. I guess empty `<Projects></Projects>` tags would work?

You can add that test in it's own `Context(){}` block at the same level as these `It()` blocks.

Reply:
Thanks @taylorsilva - I reorganized the blocks a little bit. Let me know what you think.